File: 1603. Design Parking System/Solution.py
class ParkingSystem:

    def __init__(self, big: int, medium: int, small: int):
        self.d = dict()
        self.CAP = dict()
        
        self.CAP[1] = big
        self.CAP[2] = medium
        self.CAP[3] = small
        self.d[1] = 0
        self.d[2] = 0
        self.d[3] = 0
        

    def addCar(self, carType: int) -> bool:
        if self.d[carType] <  self.CAP[carType]:
            self.d[carType] += 1
            return True
        else:
            return False
        


# Your ParkingSystem object will be instantiated and called as such:
# obj = ParkingSystem(big, medium, small)
# param_1 = obj.addCar(carType)

File: 1603. Design Parking System/Solution.cpp
class ParkingSystem {
public:
        int quota[4];
        int cur[4];
    ParkingSystem(int big, int medium, int small) {
        cur[1] = 0;
        cur[2] = 0;
        cur[3] = 0;
        quota[1] = big;
        quota[2] = medium;
        quota[3] = small;        
    }
    
    bool addCar(int carType) {
        if (cur[carType] < quota[carType])
        {
            cur[carType]++;
            return true;
        } else
        {
            return false;
        }
        
    }
};

/**
 * Your ParkingSystem object will be instantiated and called as such:
 * ParkingSystem* obj = new ParkingSystem(big, medium, small);
 * bool param_1 = obj->addCar(carType);
 */
